src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPoolIterator.cs

```diff
@@ -4,13 +4,42 @@
 using System;
 using System.Diagnostics;
 using System.Numerics;
+using System.Runtime.CompilerServices;
 using System.Threading;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Infrastructure
 {
     public struct MemoryPoolIterator
     {
-        private static readonly int _vectorSpan = Vector<byte>.Count;
+        // De Bruijn sequence https://github.com/aspnet/KestrelHttpServer/issues/1129
+        const ulong DEBRUIJN_SEQ64 = 0x03f79d71b4cb0a89;
+
+        private static readonly byte[] _debruijn64Xor =
+        {
+            0, 5, 0, 7, 6, 3, 0, 7,
+            7, 6, 5, 4, 3, 2, 0, 7,
+            6, 7, 4, 6, 6, 5, 2, 5,
+            4, 4, 3, 2, 2, 1, 0, 7,
+            5, 6, 3, 7, 5, 4, 1, 6,
+            4, 6, 2, 5, 3, 2, 1, 5,
+            3, 4, 1, 4, 2, 3, 1, 3,
+            1, 2, 1, 1, 0, 0, 0, 7,
+        };
+
+        // When this variable is set in KestrelEngine.JitReadonlyConsts it will 
+        // deterministically convert the following readonly static values to jitted consts 
+        // and embed the init code in that start up called function.
+        //
+        // This means when they are next used the have been pre-evaluated and the jit can 
+        // either directly embed them or use them for branch elimiation.
+        internal static bool StaticReadonlysJitted;
+
+        // Convert these returns to jitted consts
```

Comment:
Not sure what that would get? Each one is from a different type and I'm trying to force their evaluation rather than specifically shorten their names.